pandas/core/frame.py

```diff
@@ -140,9 +140,11 @@
 columns, the index will be passed on.
 
 Parameters
-----------%s
-right : DataFrame
+----------
+right : DataFrame or Series/dict-like object
```

Comment:
Since `Series` and `dict` are distinct types I'd separate these out as `DataFrame`, `Series` or `dict`. Update for append as well